src/app/src/test/kotlin/org/vaccineimpact/api/tests/extensions/ExpectationsTests.kt

```diff
@@ -21,12 +21,81 @@ class ExpectationsTests : MontaguTests()
         val expectations = Expectations(1,
                 2001..2080,
                 1..80,
-                CohortRestriction(null,null),
+                CohortRestriction(null, null),
                 longCountryList,
                 expectedOutcomes
         )
 
-        val result = expectations.expectedRows()
+        val result = expectations.expectedCentralRows("YF").toList()
+
+        val expectedCohorts = 80 * 80
+        val numCountries = 100
+
+        assertThat(result.count()).isEqualTo(expectedCohorts * numCountries)
+    }
+
+    @Test
+    fun `generates central row with null cohort size and outcomes`()
+    {
+        val expectedOutcomes = listOf("Dalys", "Deaths")
+
+        val longCountryList = (1..100).map {
+            Country(it.toString(), it.toString())
+        }
+        val expectations = Expectations(1,
+                2001..2080,
+                1..80,
+                CohortRestriction(null, null),
+                longCountryList,
+                expectedOutcomes
+        )
+
+        val result = expectations.expectedCentralRows("YF").toList()
+
+        assertThat(result.all { it.cohortSize == null }).isTrue()
+        assertThat(result.all { it.outcomes.all { it.value == null } }).isTrue()
+    }
```

Comment:
It would also be really good to have some tests that generate a much smaller number of rows (e.g. 2 years, 2 countries, 2 ages = 8 rows) and checks that you can get back exactly the objects expected, given that contents matters more now, rather than just count.